src/hotspot/share/code/compressedStream.hpp

```diff
@@ -81,20 +102,38 @@ class CompressedWriteStream : public CompressedStream {
   }
   void store(u_char b) {
     _buffer[_position++] = b;
+    // Writing unencoded data ends the RLE sequence of zero integers
+    _zero_count = 0;
   }
   void write(u_char b) {
     if (full()) grow();
     store(b);
   }
   void grow();
 
+  // Handle encoding of subsequent zeroes. Return true if
+  // input value is completely handled and no unsigned5 encoding required
+  bool handle_zero(juint value) {
+    if (value == 0) {
+      _zero_count = (_zero_count == 0xFF) ? 0 : _zero_count;
```

Comment:
The case of `_zero_count` overflow is not clear. Apparently, I'm missing something here.
Current code is just clearing the previously counted `_zero_count`.
I'd expect some action like storing the current number of zeros or advancing the `_position`.
Do you have a test for this?

Reply:
Good question. It works like this. On each write(), values are stored in the array, either normally or as a chain optimization (two bytes: zero byte and a number of zeros). When we get another zero value, the chain is updated. When the chain is full, we exit this function and the value is written by UNSIGNED5::write_uint_grow().

To make things clear, I have updated this code:
```diff
-      _zero_count = (_zero_count == 0xFF) ? 0 : _zero_count;
+      if (_zero_count == 0xFF) { // biggest zero chain length is 255
+        _zero_count = 1;
+        // for now, write it as an ordinary value (UNSINGED5 encodes zero int as a single byte)
+        // the new zero sequence is started if there are more than two zero values in a raw
+        return false;
+      }
```